### PR TITLE
implement thick element tracking for quads and dipoles

### PR DESCRIPTION
**Pull request by szymonlopaciuk**

### CODE REVIEW

File: xtrack/beam_elements/elements.py [hunk 1]
```diff
@@ -723,6 +723,77 @@ def get_backtrack_element(self, _context=None, _buffer=None, _offset=None):
                               _buffer=_buffer, _offset=_offset)
 
 
+class ThickCombinedFunctionDipole(BeamElement):
```
Comment (giadarol) [inline xtrack/beam_elements/elements.py]: Let's go for several elements:

- Sbend
- Rbend
- Bend
- Quadrupole
- CombinedFunctionMagnet

Need to figure out what to do about edge effects

File: xtrack/beam_elements/elements.py [hunk 2]
```diff
@@ -723,6 +723,77 @@ def get_backtrack_element(self, _context=None, _buffer=None, _offset=None):
...
+        np.array([0., 0.]),
+        mode='readonly',
+        container=self,
+    )
+
+    def get_backtrack_element(self, _context=None, _buffer=None, _offset=None):
+        ctx2np = self._buffer.context.nparray_from_context_array
+        return self.__class__(knl=-ctx2np(self.knl), _context=_context,
+                              _buffer=_buffer, _offset=_offset)
+
```
Comment (giadarol) [inline xtrack/beam_elements/elements.py]: Fo be updated and tested

File: xtrack/beam_elements/elements.py [hunk 3]
```diff
@@ -723,6 +723,77 @@ def get_backtrack_element(self, _context=None, _buffer=None, _offset=None):
...
+    knl : array
+        Normalized integrated strength of the normal components in units of m^-n.
+        Must be of length 2.
+
+    """
+    isthick = True
+
+    _xofields={
+        'knl': xo.Float64[2],
+        'hxl': xo.Float64,
```
Comment (giadarol) [inline xtrack/beam_elements/elements.py]: maybe let's store : k0, k1, h

**Review by giadarol — COMMENTED** [review]
-

**Comment by szymonlopaciuk** [comment]
Merged a long time ago.

**Pull request closed without merging**